src/multipole.F90

```diff
@@ -73,113 +77,138 @@ subroutine multipole_read(filename, multipole, i_table)
+                       multipole % end_E / 1e6_8]
+
+      ! Find the points just outside the multipole region.
+      cut1 = binary_search(nuc % energy, nuc % n_grid, insert_pts(1))
+      cut2 = binary_search(nuc % energy, nuc % n_grid, insert_pts(4)) + 1
+      if (nuc % energy(cut1) == insert_pts(1)) cut1 = cut1 - 1
+      if (nuc % energy(cut2) == insert_pts(4)) cut2 = cut2 + 1
+
+      ! Generate the new energy grid.
+      new_n_grid = nuc % n_grid - (cut2 - cut1 - 1) + 4
+      allocate(new_energy(new_n_grid))
+      new_energy(1:cut1) = nuc % energy(1:cut1)
+      new_energy(cut1+1:cut1+4) = insert_pts(:)
+      new_energy(cut1+5:new_n_grid) = nuc % energy(cut2:nuc % n_grid)
+
+      ! Compute interpolation factors for the new energy points.
+      f1 = (insert_pts(1) - nuc % energy(cut1)) &
+           / (nuc % energy(cut1+1) - nuc % energy(cut1))
+      f2 = (insert_pts(4) - nuc % energy(cut2-1)) &
+           / (nuc % energy(cut2) - nuc % energy(cut2-1))
+
+      ! Adjust the total cross section.
+      allocate(new_xs(new_n_grid))
+      new_xs(1:cut1) = nuc % total(1:cut1)
+      new_xs(cut1+1) = (ONE - f1) * nuc % total(cut1) &
+                       + f1 * nuc % total(cut1+1)
+      new_xs(cut1+2:cut1+3) = ZERO
+      new_xs(cut1+4) = (ONE - f2) * nuc % total(cut2-1) &
+                       + f2 * nuc % total(cut2)
+      new_xs(cut1+5:new_n_grid) = nuc % total(cut2:nuc % n_grid)
+      call move_alloc(new_xs, nuc % total)
+
+      ! Adjust the elastic cross section.
+      allocate(new_xs(new_n_grid))
+      new_xs(1:cut1) = nuc % elastic(1:cut1)
+      new_xs(cut1+1) = (ONE - f1) * nuc % elastic(cut1) &
+                       + f1 * nuc % elastic(cut1+1)
+      new_xs(cut1+2:cut1+3) = ZERO
+      new_xs(cut1+4) = (ONE - f2) * nuc % elastic(cut2-1) &
+                       + f2 * nuc % elastic(cut2)
+      new_xs(cut1+5:new_n_grid) = nuc % elastic(cut2:nuc % n_grid)
+      call move_alloc(new_xs, nuc % elastic)
+
+      ! Adjust the fission cross section.
+      allocate(new_xs(new_n_grid))
+      new_xs(1:cut1) = nuc % fission(1:cut1)
+      new_xs(cut1+1) = (ONE - f1) * nuc % fission(cut1) &
+                       + f1 * nuc % fission(cut1+1)
+      new_xs(cut1+2:cut1+3) = ZERO
+      new_xs(cut1+4) = (ONE - f2) * nuc % fission(cut2-1) &
+                       + f2 * nuc % fission(cut2)
+      new_xs(cut1+5:new_n_grid) = nuc % fission(cut2:nuc % n_grid)
+      call move_alloc(new_xs, nuc % fission)
+
+      ! Adjust the nu-fission cross section.
+      allocate(new_xs(new_n_grid))
+      new_xs(1:cut1) = nuc % nu_fission(1:cut1)
+      new_xs(cut1+1) = (ONE - f1) * nuc % nu_fission(cut1) &
+                       + f1 * nuc % nu_fission(cut1+1)
+      new_xs(cut1+2:cut1+3) = ZERO
+      new_xs(cut1+4) = (ONE - f2) * nuc % nu_fission(cut2-1) &
+                       + f2 * nuc % nu_fission(cut2)
+      new_xs(cut1+5:new_n_grid) = nuc % nu_fission(cut2:nuc % n_grid)
```

Comment:
It doesn't look like nu_fission has actually been calculated yet at this point, so it probably suffices to just resize nu_fission.